In [1]:
# import libraries 
import pandas as pd
from pathlib import Path
import matplotlib.pyplot as plt

## Data

This dataset can be dowloaded from : https://data.world/raghav333/fifa-players

In [14]:
#read in the CSV as a dataframe 
df = pd.read_csv('Resources/fifa_cleaned.csv')
df.head()

id          name                       full_name  birth_date  age  \
0  158023      L. Messi  Lionel Andrés Messi Cuccittini  1987-06-24   31   
1  190460    C. Eriksen    Christian  Dannemann Eriksen  1992-02-14   27   
2  195864      P. Pogba                      Paul Pogba  1993-03-15   25   
3  198219    L. Insigne                 Lorenzo Insigne  1991-06-04   27   
4  201024  K. Koulibaly               Kalidou Koulibaly  1991-06-20   27   

   height_cm  weight_kgs  positions nationality  overall_rating  ...   lwb  \
0     170.18        72.1   CF,RW,ST   Argentina              94  ...  64+2   
1     154.94        76.2  CAM,RM,CM     Denmark              88  ...  71+3   
2     190.50        83.9     CM,CAM      France              88  ...  76+3   
3     162.56        59.0      LW,ST       Italy              88  ...  63+3   
4     187.96        88.9         CB     Senegal              88  ...  73+3   

    ldm   cdm   rdm   rwb    lb   lcb    cb   rcb    rb  
0  61+2  61+2  61+2  64+2  59+2  48+2  48+2  48+2  59+2  
1  71+3  71+3  71+3  71+3  66+3  57+3  57+3  57+3  66+3  
2  77+3  77+3  77+3  76+3  74+3  72+3  72+3  72+3  74+3  
3  58+3  58+3  58+3  63+3  58+3  44+3  44+3  44+3  58+3  
4  77+3  77+3  77+3  73+3  76+3  85+3  85+3  85+3  76+3  

[5 rows x 92 columns]

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17954 entries, 0 to 17953
Data columns (total 92 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   id                            17954 non-null  int64  
 1   name                          17954 non-null  object 
 2   full_name                     17954 non-null  object 
 3   birth_date                    17954 non-null  object 
 4   age                           17954 non-null  int64  
 5   height_cm                     17954 non-null  float64
 6   weight_kgs                    17954 non-null  float64
 7   positions                     17954 non-null  object 
 8   nationality                   17954 non-null  object 
 9   overall_rating                17954 non-null  int64  
 10  potential                     17954 non-null  int64  
 11  value_euro                    17699 non-null  float64
 12  wage_euro                     17708 non-null  float64
 13  p

In [10]:
df.describe()

id           age     height_cm    weight_kgs  \
count   17954.000000  17954.000000  17954.000000  17954.000000   
mean   215411.087780     25.565445    174.946921     75.301047   
std     29758.387106      4.705708     14.029449      7.083684   
min        16.000000     17.000000    152.400000     49.900000   
25%    201117.250000     22.000000    154.940000     69.900000   
50%    222919.000000     25.000000    175.260000     74.800000   
75%    237613.500000     29.000000    185.420000     79.800000   
max    247607.000000     46.000000    205.740000    110.200000   

       overall_rating     potential    value_euro      wage_euro  \
count    17954.000000  17954.000000  1.769900e+04   17708.000000   
mean        66.240169     71.430935  2.479280e+06    9902.134628   
std          6.963730      6.131339  5.687014e+06   21995.593750   
min         47.000000     48.000000  1.000000e+04    1000.000000   
25%         62.000000     67.000000  3.250000e+05    1000.000000   
50%         66.000000     71.000000  7.000000e+05    3000.000000   
75%         71.000000     75.000000  2.100000e+06    9000.000000   
max         94.000000     95.000000  1.105000e+08  565000.000000   

       international_reputation_1_5  weak_foot_1_5  ...     penalties  \
count                  17954.000000   17954.000000  ...  17954.000000   
mean                       1.111674       2.945695  ...     48.357302   
std                        0.392168       0.663691  ...     15.810844   
min                        1.000000       1.000000  ...      5.000000   
25%                        1.000000       3.000000  ...     38.000000   
50%                        1.000000       3.000000  ...     49.000000   
75%                        1.000000       3.000000  ...     60.000000   
max                        5.000000       5.000000  ...     92.000000   

          composure       marking  standing_tackle  sliding_tackle  \
count  17954.000000  17954.000000     17954.000000    17954.000000   
mean      58.680183     47.162861        47.733040       45.705915   
std       11.625541     20.037346        21.674973       21.285812   
min       12.000000      3.000000         2.000000        3.000000   
25%       51.000000     30.000000        27.000000       24.000000   
50%       60.000000     52.500000        55.000000       52.000000   
75%       67.000000     64.000000        66.000000       64.000000   
max       96.000000     94.000000        93.000000       90.000000   

          gk_diving   gk_handling    gk_kicking  gk_positioning   gk_reflexes  
count  17954.000000  17954.000000  17954.000000    17954.000000  17954.000000  
mean      16.784449     16.556199     16.388660       16.541049     16.874067  
std       17.893895     17.108234     16.701447       17.231495     18.182943  
min        1.000000      1.000000      1.000000        1.000000      1.000000  
25%        8.000000      8.000000      8.000000        8.000000      8.000000  
50%       11.000000     11.000000     11.000000       11.000000     11.000000  
75%       14.000000     14.000000     14.000000       14.000000     14.000000  
max       90.000000     92.000000     92.000000       90.000000     94.000000  

[8 rows x 50 columns]

### Data Description
The dataframe has 17953 rows and 92 columns that has Fifa players data

In [3]:
#  Get the data types of each column.
df.dtypes

id             int64
name          object
full_name     object
birth_date    object
age            int64
               ...  
lb            object
lcb           object
cb            object
rcb           object
rb            object
Length: 92, dtype: object

In [4]:
#  Get a list of all columns within the DataFrame.
df.columns

Index(['id', 'name', 'full_name', 'birth_date', 'age', 'height_cm',
       'weight_kgs', 'positions', 'nationality', 'overall_rating', 'potential',
       'value_euro', 'wage_euro', 'preferred_foot',
       'international_reputation_1_5', 'weak_foot_1_5', 'skill_moves_1_5',
       'work_rate', 'body_type', 'release_clause_euro', 'club_team',
       'club_rating', 'club_position', 'club_jersey_number', 'club_join_date',
       'contract_end_year', 'national_team', 'national_rating',
       'national_team_position', 'national_jersey_number', 'crossing',
       'finishing', 'heading_accuracy', 'short_passing', 'volleys',
       'dribbling', 'curve', 'freekick_accuracy', 'long_passing',
       'ball_control', 'acceleration', 'sprint_speed', 'agility', 'reactions',
       'balance', 'shot_power', 'jumping', 'stamina', 'strength', 'long_shots',
       'aggression', 'interceptions', 'positioning', 'vision', 'penalties',
       'composure', 'marking', 'standing_tackle', 'sliding_tackle',
     

In [26]:
# Rename the headers to be more explanatory
df.rename(columns=str.title, inplace =True)
df

Id            Name                         Full_Name  Birth_Date  \
0      158023        L. Messi    Lionel Andrés Messi Cuccittini  1987-06-24   
1      190460      C. Eriksen      Christian  Dannemann Eriksen  1992-02-14   
2      195864        P. Pogba                        Paul Pogba  1993-03-15   
3      198219      L. Insigne                   Lorenzo Insigne  1991-06-04   
4      201024    K. Koulibaly                 Kalidou Koulibaly  1991-06-20   
...       ...             ...                               ...         ...   
17949  204322     R. McKenzie                    Rory  McKenzie  1993-10-07   
17950  239762       M. Sipľak                     Michal Sipľak  1996-02-02   
17951  235155      J. Bekkema                       Jan Bekkema  1996-04-09   
17952  244883      A. Al Yami               Abdulrahman Al Yami  1997-06-19   
17953  247187  Júnior Brumado  José Francisco dos Santos Júnior  1999-05-15   

       Age  Height_Cm  Weight_Kgs  Positions   Nationality  Overall_Rating  \
0       31     170.18        72.1   CF,RW,ST     Argentina              94   
1       27     154.94        76.2  CAM,RM,CM       Denmark              88   
2       25     190.50        83.9     CM,CAM        France              88   
3       27     162.56        59.0      LW,ST         Italy              88   
4       27     187.96        88.9         CB       Senegal              88   
...    ...        ...         ...        ...           ...             ...   
17949   25     175.26        74.8  RM,CAM,CM      Scotland              67   
17950   23     182.88        79.8         LB      Slovakia              59   
17951   22     185.42        89.8         GK   Netherlands              59   
17952   21     175.26        64.9      ST,LM  Saudi Arabia              59   
17953   19     190.50        79.8         ST        Brazil              59   

       ...   Lwb   Ldm   Cdm   Rdm   Rwb    Lb   Lcb    Cb   Rcb    Rb  
0      ...  64+2  61+2  61+2  61+2  64+2  59+2  48+2  48+2  48+2  59+2  
1      ...  71+3  71+3  71+3  71+3  71+3  66+3  57+3  57+3  57+3  66+3  
2      ...  76+3  77+3  77+3  77+3  76+3  74+3  72+3  72+3  72+3  74+3  
3      ...  63+3  58+3  58+3  58+3  63+3  58+3  44+3  44+3  44+3  58+3  
4      ...  73+3  77+3  77+3  77+3  73+3  76+3  85+3  85+3  85+3  76+3  
...    ...   ...   ...   ...   ...   ...   ...   ...   ...   ...   ...  
17949  ...  53+2  52+2  52+2  52+2  53+2  50+2  46+2  46+2  46+2  50+2  
17950  ...  57+2  55+2  55+2  55+2  57+2  57+2  58+2  58+2  58+2  57+2  
17951  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
17952  ...  41+2  35+2  35+2  35+2  41+2  39+2  32+2  32+2  32+2  39+2  
17953  ...  41+2  40+2  40+2  40+2  41+2  40+2  40+2  40+2  40+2  40+2  

[17954 rows x 92 columns]

In [38]:
#  Get a list of all columns within the DataFrame.
df.columns

Index(['Id', 'Name', 'Full_Name', 'Birth_Date', 'Age', 'Height_Cm',
       'Weight_Kgs', 'Positions', 'Nationality', 'Overall_Rating', 'Potential',
       'Value_Euro', 'Wage_Euro', 'Preferred_Foot',
       'International_Reputation_1_5', 'Weak_Foot_1_5', 'Skill_Moves_1_5',
       'Work_Rate', 'Body_Type', 'Release_Clause_Euro', 'Club_Team',
       'Club_Rating', 'Club_Position', 'Club_Jersey_Number', 'Club_Join_Date',
       'Contract_End_Year', 'National_Team', 'National_Rating',
       'National_Team_Position', 'National_Jersey_Number', 'Crossing',
       'Finishing', 'Heading_Accuracy', 'Short_Passing', 'Volleys',
       'Dribbling', 'Curve', 'Freekick_Accuracy', 'Long_Passing',
       'Ball_Control', 'Acceleration', 'Sprint_Speed', 'Agility', 'Reactions',
       'Balance', 'Shot_Power', 'Jumping', 'Stamina', 'Strength', 'Long_Shots',
       'Aggression', 'Interceptions', 'Positioning', 'Vision', 'Penalties',
       'Composure', 'Marking', 'Standing_Tackle', 'Sliding_Tackle',
     

In [34]:
df.nunique()

Id            17954
Name          16995
Full_Name     17898
Birth_Date     6156
Age              29
              ...  
Lb               97
Lcb             109
Cb              109
Rcb             109
Rb               97
Length: 92, dtype: int64

In [37]:
df['Positions'].value_counts()

CB              2243
GK              2065
ST              1747
CM               764
CDM,CM           709
                ... 
LB,RB,LM,RM        1
RM,LM,CAM,RW       1
LM,LB,ST           1
LB,LWB,LM,CB       1
CDM,RM,LM          1
Name: Positions, Length: 890, dtype: int64

In [39]:
df['Club_Position'].value_counts()

SUB    7870
RES    2885
LCB     655
RCB     654
GK      644
LB      549
RB      547
ST      449
RM      417
LM      414
RCM     413
LCM     411
CAM     308
LDM     235
RDM     234
RS      205
LS      204
CDM     147
RW      144
LW      144
CB      114
CM       80
RWB      67
LWB      66
RAM      21
LAM      20
RF       16
LF       16
CF       11
Name: Club_Position, dtype: int64